src/Page.js

```diff
@@ -209,11 +209,12 @@ function formatSiteNav(renderedSiteNav, src) {
  * @param headingIndexingLevel to generate
  */
 function generateHeadingSelector(headingIndexingLevel) {
-  let headingsSelector = 'h1';
+  let headingsSelectors = ['.index:header', 'h1'];
   for (let i = 2; i <= headingIndexingLevel; i += 1) {
-    headingsSelector += `, h${i}`;
+    headingsSelectors.push(`h${i}`);
   }
-  return headingsSelector;
+  headingsSelectors = headingsSelectors.map(selector => `${selector}:not(.noindex)`);
```

Comment:
The function comment needs to be updated since it is no longer just based on indexing level alone any more.